lib/cli-engine.js

```diff
@@ -272,6 +272,28 @@ function createIgnoreResult(filePath, baseDir) {
     };
 }
 
+/**
+ * Produces rule warnings (i.e. deprecation) from configured rules
+ * @param {Object} rules - Rules configured
+ * @param {Map} loadedRules - Map of loaded rules
+ * @returns {Object} Contains rule warnings
+ * @private
+ */
+function createRuleWarnings(rules, loadedRules) {
+    const ruleWarnings = { usedDeprecatedRules: [] };
+
+    if (rules) {
+        Object.keys(rules).forEach(name => {
+            const loadedRule = loadedRules.get(name);
+
+            if (loadedRule.meta && loadedRule.meta.deprecated) {
+                ruleWarnings.usedDeprecatedRules.push({ ruleId: name, replacedBy: loadedRule.meta.docs.replacedBy });
```

Comment:
That's a great point about `meta.docs` not necessarily existing. The "Working with Rules" page explicitly says that `meta.docs` is optional for custom rules and plugins: "In a custom rule or plugin, you can omit docs or include any properties that you need in it". 

I'm happy to move `replacedBy` up to `meta` or I can add a property check (and document it either way), but I think where `replacedBy` should hang is best decided by the maintainers?

Reply:
My personal preference would be to move it to `meta.replacedBy` rather than `meta.docs.replacedBy`, but let's see what other team members think.

cc @kaicataldo @platinumazure

Reply:
Moving `replacedBy` to `meta.replacedBy` makes sense to me :+1: